mmcls/models/backbones/gcvit.py

```diff
@@ -0,0 +1,792 @@
+            qk_scale: bool argument to scaling query, key.
+            drop_rate: dropout rate.
+            attn_drop_rate: attention dropout rate.
+            norm_layer: normalization layer.
+            layer_scale: layer scaling coefficient.
+        """
+        super().__init__()
+        if isinstance(arch, str):
+            arch = arch.lower()
+            assert arch in set(self.arch_zoo), \
+                f'Arch {arch} is not in default archs {set(self.arch_zoo)}'
+            self.arch_settings = self.arch_zoo[arch]
+            dim = self.arch_settings['dim']
+            depths = self.arch_settings['depths']
+            window_size = self.arch_settings['window_size']
+            mlp_ratio = self.arch_settings['mlp_ratio']
+            num_heads = self.arch_settings['num_heads']
+            drop_path_rate = self.arch_settings['drop_path_rate']
+            layer_scale = self.arch_settings['layer_scale']
+
+        num_features = int(dim * 2**(len(depths) - 1))
+        self.num_classes = num_classes
+        self.patch_embed = PatchEmbed(in_chans=in_chans, dim=dim)
+        self.pos_drop = nn.Dropout(p=drop_rate)
+        dpr = [
+            x.item() for x in torch.linspace(0, drop_path_rate, sum(depths))
+        ]
+        self.levels = nn.ModuleList()
+        for i in range(len(depths)):
+            level = GCViTLayer(
+                dim=int(dim * 2**i),
+                depth=depths[i],
+                num_heads=num_heads[i],
+                window_size=window_size[i],
+                mlp_ratio=mlp_ratio,
+                qkv_bias=qkv_bias,
+                qk_scale=qk_scale,
+                drop=drop_rate,
+                attn_drop=attn_drop_rate,
+                drop_path=dpr[sum(depths[:i]):sum(depths[:i + 1])],
+                norm_layer=norm_layer,
+                downsample=(i < len(depths) - 1),
+                layer_scale=layer_scale,
+                input_resolution=int(2**(-2 - i) * resolution))
+            self.levels.append(level)
+        self.norm = norm_layer(num_features)
+        self.avgpool = nn.AdaptiveAvgPool2d(1)
+        self.apply(self._init_weights)
+
+    def _init_weights(self, m):
+        if isinstance(m, nn.Linear):
+            trunc_normal_(m.weight, std=.02)
+            if isinstance(m, nn.Linear) and m.bias is not None:
+                nn.init.constant_(m.bias, 0)
+        elif isinstance(m, nn.LayerNorm):
+            nn.init.constant_(m.bias, 0)
+            nn.init.constant_(m.weight, 1.0)
+
+    @torch.jit.ignore
+    def no_weight_decay_keywords(self):
+        return {'rpb'}
+
+    def forward_features(self, x):
+        x = self.patch_embed(x)
+        x = self.pos_drop(x)
+
+        for level in self.levels:
+            x = level(x)
+
+        x = self.norm(x)
+        x = _to_channel_first(x)
+        x = self.avgpool(x)
+        x = torch.flatten(x, 1)
+        return x
+
+    def forward(self, x):
```

Comment:
The `forward` and `forward_features` should be merged into one function. Also, add `out_indices` option in forward is welcomed, you can refer https://github.com/open-mmlab/mmclassification/blob/51b7c754e73e0e20479420ca664eed69796d327b/mmcls/models/backbones/convnext.py#L301 for more details. This design is friendly for the downstream tasks, like detection and segmentation